include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -241,14 +193,16 @@ __pattern_transform_scan_base(_ExecutionPolicy&& __exec, _Iterator1 __first, _It
         oneapi::dpl::__par_backend_hetero::__internal::__buffer<_NewExecutionPolicy, _Type> __tmp_buf(__policy, __n);
         auto __first_tmp = __tmp_buf.get();
         auto __last_tmp = __first_tmp + __n;
+        auto __keep2 = oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::write, _Iterator2>();
+        auto __buf2 = __keep2(__first_tmp, __last_tmp);
 
         // Run main algorithm and save data into temporary buffer
-        __pattern_transform_scan_base_impl(__policy, __first, __last, __first_tmp, __unary_op, __init, __binary_op,
-                                           _Inclusive{});
+        oneapi::dpl::__par_backend_hetero::__parallel_transform_scan(__policy, __buf1.all_view(), __buf2.all_view(), __n, __unary_op, __init, __binary_op,
+                                           _Inclusive{}).wait();
 
         // Move data from temporary buffer into results
         oneapi::dpl::__internal::__pattern_walk2_brick(
-            ::std::forward<_NewExecutionPolicy>(__policy), __first_tmp, __last_tmp, __result,
+            ::std::move(__policy), __first_tmp, __last_tmp, __result,
```

Comment:
Why do we do such change?

Reply:
`__policy` is created a few lines above, so forwarding it does not make sense because it is not passed in as a parameter. It is not used after is line, so a `move` here should be fine.